docs/src/main/asciidoc/panache-rest.adoc

```diff
@@ -0,0 +1,242 @@
+_Do not implement these interfaces and don't provide custom methods because they will be ignored._ You can, however, override the methods from the extended interface in order to customize them (see the section at the end).
+
+The generated resources will be functionally equivalent for both entities and repositories.
+The only difference being the particular data access pattern in use.
+
+Panache REST will generate a JAX-RS resource class similar to this:
+
+[source,java]
+----
+@Path("/entities") // Base bath is a pluralised name of an entity
+public class EntityCrudResourceImpl implements EntityCrudResource { // The actual class name is going to be unique
+    @GET
+    @Produces("application/json")
+    @Path("{id}")
+    public Entity get(@PathParam("id") ID id){
+        // ...
+    }
+
+    @GET
+    @Produces("application/json")
+    public List<Entity> list(){
+        // ...
+    }
+
+    @Transactional
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    public Response add(Entity entity) {
+        // ..
+    }
+
+    @Transactional
+    @PUT
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("{id}")
+    public Response update(@PathParam("id") ID id, Entity entity) {
+        // ..
+    }
+
+    @Transactional
+    @DELETE
+    @Path("{id}")
+    public void delete(@PathParam("id") ID id) {
+        // ..
+    }
+}
+----
+
+=== PanacheEntityCrudResource
+
+If your application has an entity (e.g. `Person`) that extends either `PanacheEntity` or `PanacheEntityBase` class, you could instruct Panache REST to generate its CRUD resource with the following interface:
+
+[source,java]
+----
+public interface PersonCrudResource extends PanacheEntityCrudResource<Person, Long> {
+}
+----
+
+=== PanacheRepositoryCrudResource
+
+If your application has a simple entity (e.g. `Person`) and a repository (e.g. `PersonRepository`) that implements either `PanacheRepository` or `PanacheRepositoryBase` interface, you could instruct Panache REST to generate its CRUD resource with the following interface:
+
+[source,java]
+----
+public interface PersonCrudResource extends PanacheRepositoryCrudResource<PersonRepository, Person, Long> {
+}
+----
+
+== Resource customisation
+
+Panache REST provides a `@PanacheRestResource` annotation that can be used to customize certain features of the resource.
+It can be used on your CRUD resource interface:
+
+[source,java]
+----
+@PanacheRestResource(hal = true)
+public interface PersonCrudResource extends PanacheEntityCrudResource<Person, Long> {
+    @PanacheRestResource(exposed = false)
+    void delete(Long id);
+}
+----
+
+=== Available options
+
+* `hal` - in addition to the standard `application/json` responses, generates additional methods that can return `application/hal+json` responses if requested via `Accept` header.
```

Comment:
Wondering if we should have an array of content types instead with the default being `{ "application/json" }`. That would be much more flexible.

Reply:
/cc @FroMage

Reply:
Well, I don't think we currently support any response type, but this seems like a valid point for the future.

Reply:
`application/json` is hardcoded and `application/hal+json` can be enabled as an extra. Because HAL is a complex format I had to implement a custom serialiser for it which serialisers special wrapper objects.

Reply:
OK, let's keep it that way for now.